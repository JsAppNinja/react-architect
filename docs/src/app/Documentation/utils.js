export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
