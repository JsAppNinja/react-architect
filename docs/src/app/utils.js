import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
