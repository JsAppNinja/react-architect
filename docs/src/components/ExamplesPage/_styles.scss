@import '~globals';

.examples-page__card__description {
  margin-bottom: 16px;
}

.examples-page__example-link {
  border: 0;
  left: -1000px;
  outline: none;
  padding: 0;
  position: fixed;
  top: -1000px;

  &-btn {
    opacity: 0;
    transform: rotate3d(0, 0, 1, 90deg);

    &--active {
      opacity: 1;
    }

    &.md-btn {
      transition-duration: .5s;
      transition-property: background, opacity;
    }

    @media (max-width: #{$md-desktop-min-width - 1}) {
      display: none;
    }
  }
}
