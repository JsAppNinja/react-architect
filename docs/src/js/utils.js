import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
