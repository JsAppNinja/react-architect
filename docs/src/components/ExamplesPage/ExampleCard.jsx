import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Card from 'react-md/lib/Cards/Card';
import CardTitle from 'react-md/lib/Cards/CardTitle';
import CardText from 'react-md/lib/Cards/CardText';
import { toCaterpillarCase } from 'utils/strings';

import Markdown from 'components/Markdown';

const ExampleCard = ({ title, description, code, children: propChildren, tableCard, className, ...props }) => {
  const markdown = `
\`\`\`jsx
${code}
\`\`\`
`;

  let descriptionMarkdown;
  if (description) {
    descriptionMarkdown = (
      <Markdown
        key="description"
        component="div"
        markdown={description}
        className="md-text-container examples-page__card__description"
      />
    );
  }

  let children = propChildren;
  if (tableCard && description) {
    children = [
      <CardText key="description">{descriptionMarkdown}</CardText>,
      React.cloneElement(children, { key: 'table-card-example' }),
    ];
  } else if (!tableCard) {
    children = <CardText key="example-card-text">{descriptionMarkdown}{children}</CardText>;
  }

  return (
    <Card
      id={toCaterpillarCase(title)}
      {...props}
      tabIndex={-1}
      tableCard={tableCard}
      className={cn('md-cell md-cell--12', className)}
      expanderIconChildren="code"
      expanderTooltipLabel="View the source for this example."
      expanderTooltipDelay={300}
    >
      <CardTitle expander title={title} />
      <CardText expandable>
        <Markdown markdown={markdown} />
      </CardText>
      {children}
    </Card>
  );
};

ExampleCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  code: PropTypes.string.isRequired,
  children: PropTypes.node,
  tableCard: PropTypes.bool,
  className: PropTypes.string,
};

export default ExampleCard;
