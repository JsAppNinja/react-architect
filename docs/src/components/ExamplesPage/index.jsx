import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { GridList } from 'react-md';

import Markdown from 'components/Markdown';
import withMinHeight from 'components/hoc/withMinHeight';

import './_styles.scss';
import ExampleCard from './ExampleCard';

const ExamplesPage = ({
  style,
  className,
  description,
  examples,
  children,
}) => {
  const cards = examples.map(example => <ExampleCard key={example.title} {...example} />);

  return (
    <GridList
      component="section"
      gutter={40}
      spacing={16}
      size={12}
      style={style}
      className={cn('examples-page', className)}
    >
      <Markdown component="header" className="md-text-container" markdown={description} />
      {cards}
      {children}
    </GridList>
  );
};

ExamplesPage.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  description: PropTypes.string,
  examples: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    code: PropTypes.string,
    description: PropTypes.string,
    children: PropTypes.node,
    tableCard: PropTypes.bool,
  })).isRequired,
  children: PropTypes.node,
};

export default withMinHeight(ExamplesPage);
