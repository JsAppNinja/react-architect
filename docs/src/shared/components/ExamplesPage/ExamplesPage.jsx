import React, { PureComponent, PropTypes } from 'react';
import cn from 'classnames';

import Markdown from 'components/Markdown';
import ExampleCard from './ExampleCard';

export default class ExamplesPage extends PureComponent {
  static propTypes = {
    style: PropTypes.object,
    className: PropTypes.string,
    params: PropTypes.shape({
      component: PropTypes.string.isRequired,
      section: PropTypes.string,
    }).isRequired,
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
    }).isRequired,

    children: PropTypes.node,
  };

  constructor(props) {
    super(props);

    this.state = { examples: [], readme: '' };
  }

  componentWillMount() {
    const { component, section } = this.props.params;
    let folder = component;
    if (component === 'selection-control') {
      folder = section;
    } else if (section) {
      folder = `${section}/${component}`;
    }

    if (__CLIENT__) {
      require.ensure([], require => {
        const examples = require(`components/ReactMD/${folder}/index.js`).default;
        const readme = require(`components/ReactMD/${folder}/README.md`);

        this.setState({ examples, readme });
      });
    } else {
      const examples = require(`components/ReactMD/${folder}/index.js`).default;
      const readme = require(`components/ReactMD/${folder}/README.md`);

      this.setState({ examples, readme });
    }
  }

  render() {
    const { readme } = this.state;
    const { style, className } = this.props;
    const examples = this.state.examples.map((example, i) => (
      <ExampleCard key={example.title || i} {...example} fallbackId={`example-${i}`} />
    ));

    return (
      <section style={style} className={cn('examples-page md-grid md-grid--40-16', className)}>
        <Markdown markdown={readme} component="header" className="md-cell md-cell--12 md-text-container" />
        {examples}
      </section>
    );
  }
}
