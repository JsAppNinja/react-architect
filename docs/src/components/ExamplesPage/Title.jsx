import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { CardTitle } from 'react-md';

import { QuickLinkTitle } from 'components/QuickLink';

@connect(({ media: { desktop } }) => ({ desktop }))
export default class Title extends PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    desktop: PropTypes.bool,

    // This is only required because of how the Card reads the prop types for this to work.
    expander: PropTypes.bool,
  };

  static defaultProps = {
    expander: true,
  };

  render() {
    const { id, expander, title, desktop } = this.props;

    return (
      <CardTitle
        title={<QuickLinkTitle id={id} title={title} desktop={desktop} />}
        expander={expander}
        className="quick-link"
      />
    );
  }
}
