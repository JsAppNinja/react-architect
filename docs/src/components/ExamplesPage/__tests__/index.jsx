/* eslint-env jest */
import React from 'react';
import { renderReduxRouterSnapshot } from 'utils/testing';

import ExamplesPage from '../';

describe('ExamplesPage', () => {
  it('should render correctly', () => {
    const examples = [{
      title: 'First Example',
      description: 'Something that has some **markdown** and __other stuff__.',
      code: '<h1>Wowwwwwwwwwwwwww</h1>',
      children: <h1>Wowwwwwwwwwwwwww</h1>,
    }];

    const props = {
      examples,
      description: `
      ## Some Example page
      That has content

      And some amazing **markdown**!

      - one
      - two
      - three
      `,
    };
    const tree = renderReduxRouterSnapshot(<ExamplesPage {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
