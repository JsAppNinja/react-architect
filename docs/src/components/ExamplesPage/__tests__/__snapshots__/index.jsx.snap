// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExamplesPage should render correctly 1`] = `
<section
  class="md-grid md-grid--40-16 examples-page"
>
  <header
    class="markdown-container md-text-container md-cell md-cell--12"
  >
    <pre>
      <code>
          ## Some Example page
  That has content

  And some amazing **markdown**!

  - one
  - two
  - three

      </code>
    </pre>
  </header>
  <div
    class="md-paper md-paper--1 md-card md-background--card md-cell md-cell--12"
    id="first-example"
    tabindex="-1"
  >
    <div
      class="md-card-title md-card-title--primary quick-link"
    >
      <h2
        class="md-card-title--title md-card-title--large md-text"
      >
        <div
          class="quick-link__container"
        >
          First Example
          <a
            class="quick-link__link quick-link__link--title"
            href="#first-example"
          >
            <i
              class="md-icon material-icons"
            >
              link
            </i>
          </a>
        </div>
      </h2>
      <button
        class="md-btn md-btn--icon md-pointer--hover md-btn--tooltip md-inline-block md-collapser md-collapser--card"
        type="button"
      >
        <div
          class="md-ink-container"
        >
          <span
            aria-hidden="true"
          />
        </div>
        <div
          class="md-tooltip-container"
        >
          <span
            aria-hidden="true"
          />
        </div>
        <svg
          class="md-icon md-text--inherit"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
    </div>
    <section
      class="md-card-text"
    >
      <div
        class="markdown-container examples-page__card__description md-text-container"
      >
        <p>
          Something that has some 
          <strong>
            markdown
          </strong>
           and 
          <strong>
            other stuff
          </strong>
          .
        </p>
        

      </div>
      <h1>
        Wowwwwwwwwwwwwww
      </h1>
    </section>
  </div>
</section>
`;
